turbopack/crates/turbopack-ecmascript/src/lib.rs

```diff
@@ -143,6 +143,14 @@ pub struct EcmascriptOptions {
     /// If false, they will reference the whole directory. If true, they won't
     /// reference anything and lead to an runtime error instead.
     pub ignore_dynamic_requests: bool,
+    /// The list of export names that should make tree shaking bail off. This is
+    /// required because tree shaking can split imports like `export const
+    /// runtime = 'edge'` as a separate module.
+    ///
+    /// Currently the analysis of these exports are statically verified by `NextPageStaticInfo`,
+    /// which is a `CustomTransformer` implementation and we don't have a way to apply it after
+    /// tree shaking.
```

Comment:
The reason is weird to me. If that analysis is applied before tree shaking, it should see exactly the same code with and without tree shaking... Why would opting out tree shaking make a difference?

Reply:
I tried, and I think I was wrong. It was simply failing due to post-build analysis

Reply:
We hit

```

Page: {"type":"app","side":"server","page":"/export/basic/page"}
./app/export/inherit/page.tsx
Invalid page configuration
Next.js can't recognize the exported `preferredRegion` field in "app/export/inherit/page.tsx" as it was not assigned to a string literal or an array of string literals.
The default runtime will be used instead.

```

anyway, because next.js tries to analyze pages from JS.

Reply:
It should hit that error. The test case is testing for that.